Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -709,44 +811,71 @@ namespace OpenInfraPlatform {
 				}
 #endif 
 
-				// (8/8) IfcTrimmedCurve SUBTYPE OF IfcBoundedCurve
+				// IfcTrimmedCurve SUBTYPE OF IfcBoundedCurve
 				/**********************************************************************************************/
 				/*! \brief Calculates trimming points of the curve.
 				* \param[in] polycurve				A pointer to data from c\ IfcTrimmedCurve.
 				* \param[out] targetVec				The tessellated line.
 				* \param[out] segmentStartPoints	The starting points of separate segments.
 				*/
-				void convertIfcTrimmedCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcTrimmedCurve>& trimmed_curve,
+				void convertIfcTrimmedCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcTrimmedCurve>& trimmedCurve,
 					std::vector<carve::geom::vector<3>>& targetVec,
 					std::vector<carve::geom::vector<3>>& segmentStartPoints) const throw(...)
 				{
-					std::shared_ptr<typename IfcEntityTypesT::IfcCurve> basis_curve = trimmed_curve->BasisCurve.lock();
-					std::vector<carve::geom::vector<3> > basis_curve_points;
+					// **************************************************************************************************************************
+					//	https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC2/HTML/schema/ifcgeometryresource/lexical/ifctrimmedcurve.htm
+					//	ENTITY IfcTrimmedCurve
+					//		SUBTYPE OF(IfcBoundedCurve);
+					//			BasisCurve: IfcCurve;
+					//			Trim1: SET[1:2] OF IfcTrimmingSelect;
+					//			Trim2: SET[1:2] OF IfcTrimmingSelect;
+					//			SenseAgreement: IfcBoolean;
+					//			MasterRepresentation: IfcTrimmingPreference;
+					//		WHERE
+					//			Trim1ValuesConsistent : (HIINDEX(Trim1) = 1) OR(TYPEOF(Trim1[1]) < > TYPEOF(Trim1[2]));
+					//			Trim2ValuesConsistent: (HIINDEX(Trim2) = 1) OR(TYPEOF(Trim2[1]) < > TYPEOF(Trim2[2]));
+					//			NoTrimOfBoundedCurves: NOT('IFC4X1.IFCBOUNDEDCURVE' IN TYPEOF(BasisCurve));

```

Comment:
Because of this WHERE rule, all functions interpreting a bounded curve (or its derivative) do not need to react to trimming parameters. Rather, if they are there, an incosistent modelling exception should be thrown.

Reply:
Resolved: 475dbf1